test/external_tests.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+from argparse import ArgumentParser, Namespace
+import os
+from os import listdir
+from pathlib import Path
+import sys
+
+# Our scripts/ is not a proper Python package so we need to modify PYTHONPATH to import from it
+# pragma pylint: disable=import-error,wrong-import-position
+SCRIPTS_DIR = Path(__file__).parents[1] / "scripts"
+sys.path.insert(0, str(SCRIPTS_DIR))
+from common.shell_command import run_cmd
+
+EXTERNAL_TESTS_DIR = Path(__file__).parent / "externalTests"
+
+
+class OptionNotFound(Exception):
+    pass
+
+
+class ExternalTestNotFound(Exception):
+    pass
+
+
+def external_tests_scripts() -> dict:
+    test_scripts: dict = {}
+    for f in listdir(EXTERNAL_TESTS_DIR):
+        file_path = Path(EXTERNAL_TESTS_DIR) / f
+        if file_path.is_file():
+            file_name, extension = file_path.name.split(".")
+            # TODO: Remove `file_name == "common"` when we complete the migration
+            # of the external tests to python, since there will be no more
+            # common.sh script in the externalTests folder.
+            if extension == "sh" and not file_name == "common":
+                test_scripts[file_name] = file_path
+    return test_scripts
```

Comment:
The pathlib equivalent of `listdir()` is [`Path.iterdir()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.iterdir) and `Path` has members for the extension and the stem.

This also would be much simpler as a dict comprehension:

```suggestion
def external_tests_scripts() -> dict:
    # TODO: Remove `file_path.stem != "common"` when we complete the migration
    # of the external tests to python, since there will be no more
    # common.sh script in the externalTests folder.
    return {
        file_path.stem: file_path
        for file_path in Path(EXTERNAL_TESTS_DIR).iterdir()
        if file_path.is_file() and file_path.suffix == "sh" and file_path.stem != "common"
    }